src/common_node.cpp

```diff
@@ -19,3 +20,85 @@ void CommonNode::heartbeat_timer_callback() {
     heartbeat_publisher->publish(message);
     RCLCPP_DEBUG(this->get_logger(), "CommonNode::heartbeat_timer_callback: Published heartbeat message with sender_id: %s, tick: %u, active: %d", this->get_fully_qualified_name(), message.tick, message.active);
 }
+
+/**
+ * @brief Handles the completion of a job.
+ *
+ * This function sends a job_finished message with the given error code and payload.
+ * Additionally, deactivates the node.
+ *
+ * @param error_code The error code associated with the job completion (EXIT_SUCCESS == 0 if no error).
+ * @param payload The payload data associated with the job completion.
+ */
+void CommonNode::job_finished(const uint8_t error_code, const nlohmann::json &payload)
+{
+    interfaces::msg::JobFinished msg;
+
+    msg.sender_id = this->get_fully_qualified_name();
+    msg.error_code = error_code;
+    msg.payload = payload.dump();
+
+    job_finished_publisher->publish(msg);
+    RCLCPP_DEBUG(this->get_logger(), "CommonNode::job_finished: Sent job_finished message with custom error_code and payload");
```

Comment:
Die Logging macros haben auch eine printf ähnliche Funktionalität.
Vielleicht will man hier zumindest den error code ausgeben.
```suggestion
    RCLCPP_DEBUG(this->get_logger(), "CommonNode::job_finished: Sent job_finished message with custom error_code: %" PRIu8 " and payload", error_code);
```

Oder:

```suggestion
    RCLCPP_DEBUG(this->get_logger(), "CommonNode::job_finished: Sent job_finished message with custom error_code: %" PRIu8 " and payload: \n%s", error_code, payload.dump(4).c_str());
```

Reply:
Das würde dann natürlich auch die anderen job_finished überladungen betreffen.

Reply:
Es gibt btw. auch eine std::cout ähnliche Variante der Logging-Nachrichten.

Reply:
printf ftw!!!

Reply:
Ich hätte es beim error_code belassen, damit der Debug output nicht zu stark überladen wird